src/Altinn.Apps/AppTemplates/AspNet/Altinn.App.PlatformServices/Implementation/ProcessChangeHandler.cs

```diff
@@ -0,0 +1,426 @@
+                    NewProcessState = startState,
+                    Events = events,
+                };
+            }
+
+            return null;
+        }
+
+        private static InstanceEvent GenerateProcessChangeEvent(string eventType, Instance instance, DateTime now, ClaimsPrincipal user)
+        {
+            InstanceEvent instanceEvent = new InstanceEvent
+            {
+                InstanceId = instance.Id,
+                InstanceOwnerPartyId = instance.InstanceOwner.PartyId,
+                EventType = eventType,
+                Created = now,
+                User = new PlatformUser
+                {
+                    UserId = user.GetUserIdAsInt(),
+                    AuthenticationLevel = user.GetAuthenticationLevel(),
+                    OrgId = user.GetOrg()
+                },
+                ProcessInfo = instance.Process,
+            };
+
+            return instanceEvent;
+        }
+
+        private static InstanceEvent CopyInstanceEventValue(InstanceEvent e)
+        {
+            return new InstanceEvent
+            {
+                Created = e.Created,
+                DataId = e.DataId,
+                EventType = e.EventType,
+                Id = e.Id,
+                InstanceId = e.InstanceId,
+                InstanceOwnerPartyId = e.InstanceOwnerPartyId,
+                ProcessInfo = new ProcessState
+                {
+                    Started = e.ProcessInfo?.Started,
+                    CurrentTask = new ProcessElementInfo
+                    {
+                        Flow = e.ProcessInfo?.CurrentTask.Flow,
+                        AltinnTaskType = e.ProcessInfo?.CurrentTask.AltinnTaskType,
+                        ElementId = e.ProcessInfo?.CurrentTask.ElementId,
+                        Name = e.ProcessInfo?.CurrentTask.Name,
+                        Started = e.ProcessInfo?.CurrentTask.Started,
+                        Ended = e.ProcessInfo?.CurrentTask.Ended,
+                        Validated = new ValidationStatus
+                        {
+                            CanCompleteTask = e.ProcessInfo?.CurrentTask?.Validated?.CanCompleteTask ?? false,
+                            Timestamp = e.ProcessInfo?.CurrentTask?.Validated?.Timestamp
+                        }
+                    },
+
+                    StartEvent = e.ProcessInfo?.StartEvent
+                },
+                User = new PlatformUser
+                {
+                    AuthenticationLevel = e.User.AuthenticationLevel,
+                    EndUserSystemId = e.User.EndUserSystemId,
+                    OrgId = e.User.OrgId,
+                    UserId = e.User.UserId
+                }
+            };
+        }
+
+        /// <summary>
+        /// Moves instance's process to nextElement id. Returns the instance together with process events.
+        /// </summary>
+        public ProcessStateChange ProcessNext(Instance instance, string nextElementId, ClaimsPrincipal userContext)
```

Comment:
I don't like the try catch in this method. I think it is safer to let an exception terminate the process. I'm not sure what kind of state we get with the current logic.

1. In the case of an exception, this method returns null.
2. `ProcessChangeHandler.HandleMoveToNext` checks the returned value and if it is null nothing is done, apart from setting `processChange.ProcessStateChange` to null.
3. `ProcessEngine.Next` does nothing. 
4. `ProcessController.NextElement` checks `processChangeContext.FailedProcessChange`, but the value is probably still false?
5. `ProcessController.NextElement` returns Ok.

Reply:
removed try catch